File: problem_2/solution.rb
def sum_even_fibonacci_numbers
  sum = 0
  first_item = 1
  second_item = 1
  third_item = first_item+second_item
  while third_item < 4000000
    sum += third_item
    first_item = second_item + third_item
    second_item = first_item + third_item
    third_item = first_item + second_item
  end
  sum
end

before = Time.now
result = sum_even_fibonacci_numbers
elapsed = '%.6fs' % (Time.now - before).to_s

puts "=> Result: #{result}"
puts  "=> Time: #{elapsed}"

File: problem_2/solution.py
import time


def sum_even_fibonacci_numbers_1():
    f1, f2, s, = 0, 1, 0,
    while f2 < 4000000:
        f2, f1 = f1, f1 + f2
        if f2 % 2 == 0:
            s += f2
    return s


def sum_even_fibonacci_numbers_2():
    s, a, b = 0, 1, 1
    c = a + b
    while c < 4000000:
        s += c
        a = b + c
        b = a + c
        c = a + b
    return s
t1 = time.time()
output = sum_even_fibonacci_numbers_1()
t2 = time.time()

print "=> Solution 1"
print "=> Result: %s" % output
print "=> Time: %fs" % (t2 - t1)

t1 = time.time()
output = sum_even_fibonacci_numbers_2()
t2 = time.time()

print "=> Solution 2"
print "=> Result: %s" % output
print "=> Time: %fs" % (t2 - t1)
